tests/gso.test.ts

```diff
@@ -0,0 +1,32 @@
+import dotenv from 'dotenv';
+
+import { expect } from '@jest/globals';
+import { clusterApiUrl } from '@solana/web3.js';
+import { GSO } from '../src';
+
+dotenv.config();
+
+describe('gso-sdk', () => {
+  const client = new GSO(
+    process.env.RPC_URL || clusterApiUrl('devnet'),
+    'confirmed',
+  );
+
+  it('can fetch gsos', async () => {
+    const gsos = await client.getGsos();
+    console.info(gsos);
+
+    expect(gsos).toBeTruthy();
+    expect(gsos).toBeInstanceOf(Array);
+    expect(gsos.length).toBeGreaterThanOrEqual(1);
+    gsos.forEach((gso) => {
+      expect(gso.soName).toBeTruthy();
+      expect(gso.strike).toBeTruthy();
+      expect(Number(gso.expiration) * 1000).toBeGreaterThan(Date.now());
```

Comment:
The filtering in the code is for subscription period end, not expiration. The fact that subscription <= expiration is an implementation detail that while it should be safe, isnt guaranteed in the client implementation